azure/services/aso/tags_test.go

```diff
@@ -151,4 +155,31 @@ func TestReconcileTags(t *testing.T) {
 		g.Expect(errors.As(err, &recerr)).To(BeTrue())
 		g.Expect(recerr.IsTransient()).To(BeTrue())
 	})
+
+	// This verifies the case where some other mechanism like Azure Policy is adding more tags that show up in
+	// the status without first appearing in the spec.
+	t.Run("existing tags contain more than spec", func(t *testing.T) {
+		g := NewWithT(t)
+
+		mockCtrl := gomock.NewController(t)
+		tag := mock_aso.NewMockTagsGetterSetter[*asoresourcesv1.ResourceGroup](mockCtrl)
+
+		existing := &asoresourcesv1.ResourceGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					asoannotations.ReconcilePolicy: string(asoannotations.ReconcilePolicyManage),
+				},
+			},
+		}
+		tag.EXPECT().GetActualTags(existing).Return(infrav1.Tags{"existing": "value", "extra": "tag"})
+		tag.EXPECT().GetDesiredTags(existing).Return(infrav1.Tags{"existing": "value"})
+		tag.EXPECT().GetAdditionalTags().Return(nil)
+
+		parameters := &asoresourcesv1.ResourceGroup{}
+		tag.EXPECT().GetDesiredTags(parameters).Return(nil)
```

Comment:
Why is `GetDesiredTags()` returning nil here? I think I'm a bit confused about getting tags from existing vs parameters and also why we're setting tags to the actual instead of the desired in the line below.